src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,134 @@
+import "./Carousel.css";
+import React, { ReactElement, useEffect, useRef, useState } from "react";
+import { LEFT, RIGHT, GAP } from "../../constants/constant";
+import CarouselButton from "./CarouselButton";
+
+const CLEAREANCE_RIGHT: number = 300;
+
+interface IButtonVisiblity {
+  left: boolean;
+  right: boolean;
+}
+
+const Carousel = (): ReactElement => {
+  const carouselItem = useRef<HTMLDivElement>(null);
+  const carouselRef = useRef<HTMLDivElement>(null);
+  const carouselContainerRef = useRef<HTMLDivElement>(null);
+
+  const [btnVisible, setbtnVisble] = useState<IButtonVisiblity>({
+    left: false,
+    right: true,
+  });
+
+  const { left, right } = btnVisible;
+
+  const checkVisiblity = (direction: string) => {
+    const carouselElement = carouselRef.current;
+    if (!carouselElement) {
+      return;
+    }
+
+    if (
+      direction === RIGHT &&
+      carouselElement.scrollWidth - CLEAREANCE_RIGHT <=
+        carouselElement.clientWidth + carouselElement.scrollLeft
+    ) {
+      setbtnVisble({
+        right: false,
+        left: true,
+      });
+      return;
+    }
+
+    if (direction === LEFT && carouselElement.scrollLeft === 0) {
+      setbtnVisble({
+        right: true,
+        left: false,
+      });
+      return;
+    }
+
+    setbtnVisble({
+      right: true,
+      left: true,
+    });
+  };
+
+  const scrollCarousel = (direction: string) => {
+    if (!carouselItem.current || !carouselRef.current) {
+      return;
+    }
+    let width = Math.floor(carouselItem.current.scrollWidth) + GAP;
+    width = direction === LEFT ? -width : width;
+
+    carouselRef.current.scrollBy({
+      left: width,
+      behavior: "smooth",
+    });
+
+    checkVisiblity(direction);
+  };
+
+  useEffect(() => {
+    const carouselObserver = new IntersectionObserver(
+      ([entry]) => {
+        if (entry.isIntersecting) {
+          const header = document.querySelector(".navbar-wrapper")!;
+          const filterSection = document.querySelector(".filter-wrapper")!;
+          header.classList.remove("hide-header");
+          filterSection.classList.remove("filter-sticky");
+        }
+      },
+      {
+        root: null,
+        threshold: [0, 1],
+      }
+    );
+    carouselObserver.observe(carouselContainerRef.current!);
+    return () => carouselObserver.disconnect();
+  }, []);
```

Comment:
Carouse's job should be only to render Carousel. It should not be aware of header and filer selection.